test/run

```diff
@@ -2,7 +2,7 @@
 
 # Default Python Versions
 # shellcheck source=bin/default_pythons
-source "$BIN_DIR/default_pythons"
+source "bin/default_pythons"
```

Comment:
was this on purpose?

Reply:
It was, the file wasn't being found before the change. should I have changed the shellcheck instead?

Reply:
oh, so this just never worked before?

Reply:
Yep, missed its effects in this test script due to false positives. I checked that it worked this time by running tests for all stacks with only one binary generated, it responds correctly to the failure case now